src/randoop/operation/ConstructorCall.java

```diff
@@ -216,31 +220,68 @@ public ExecutionOutcome execute(Object[] statementInput, PrintStream out) {
   }
 
   /**
-   * Returns the return type of this constructor.
+   * {@inheritDoc}
+   * @return type of the object created (i.e., class for constructor).
    */
+  @Override
   public Class<?> getOutputType() {
-    if (outputTypeCached == null) {
```

Comment:
I just went ahead and saved the type into the variable rather than checking every call whether it had been saved into the variable.